src/terminal/parser/InputStateMachineEngine.cpp

```diff
@@ -308,6 +308,26 @@ bool InputStateMachineEngine::ActionCsiDispatch(const wchar_t wch,
     const unsigned short cRemainingArgs = (cParams >= 1) ? cParams - 1 : 0;
 
     bool fSuccess = false;
+
+    // Handle "Intermediate" characters, if any
+    if (cIntermediate > 0)
+    {
+        switch (wchIntermediate)
+        {
+        case CsiIntermediateCodes::MOUSE_SGR:
+        {
+            DWORD dwButtonState = 0;
+            DWORD dwEventFlags = 0;
+            dwModifierState = _GetSGRMouseModifierState(rgusParams, cParams);
+            fSuccess = _GetXYPosition(rgusParams, cParams, &row, &col);
+            fSuccess = _UpdateSGRMouseButtonState(wch, rgusParams, &dwButtonState, &dwEventFlags) && fSuccess;

```

Comment:
So, I hate this. I need it to be in this order because we _always_ need to update our MouseButtonState (regardless of failure).

If I swap the order and `_GetXYPosition()` returns false, we would never update the state.

Reply:
okay that's _insane_ but I can deal with it